main/src/test/scala/com/dwolla/fs2aws/s3/S3ClientSpec.scala

```diff
@@ -21,41 +24,72 @@ class S3ClientSpec(implicit ee: ExecutionEnv) extends Specification with Matcher
   val key = tagKey("key")
   val expectedContentType = "text/x-shellscript"
 
-  val capturedPutObjectRequest = IO(capture[PutObjectRequest])
-  val capturedS3ProgressListener = IO(capture[S3ProgressListener])
   val mockTransferManager = IO(mock[TransferManager])
   val objectMetadata = IO {
     val om = new ObjectMetadata()
     om.setContentType(expectedContentType)
     om
   }
 
+  implicit val cs = IO.contextShift(ee.executionContext)
+
   "S3Client" should {
     "upload the bytes of the stream to the given S3 location" >> {
       for {
-        putCaptor ← capturedPutObjectRequest
-        progressListenerCaptor ← capturedS3ProgressListener
-        tm ← mockTransferManager.flatTap(tm ⇒ IO {
-          tm.upload(putCaptor.capture, progressListenerCaptor.capture) returns null
+        deferredUploadArguments <- Deferred[IO, (PutObjectRequest, S3ProgressListener)]
```

Comment:
@lochnesh Since you approved this PR, we merged #4, so this test was added and needed to be updated with the new interface. You may want to take another look before I merge this next week.

Reply:
lgtm